webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -103,32 +97,37 @@ const ProjectDetails = memo((projectDetails: ProjectDetails) => {
         },
     });
 
-    // Reset the state when the component is unmounted
-    useEffect(() => {
-        if (!getWebsocketEventState(state).isSubscriptionDeleted) {
-            fetchSubscriptionList();
+    const handleToggleShowAllSubscriptions = (showAllSubscriptions: boolean) => {
+        if (showAllSubscriptions) {
+            setSubscriptionListApiParams({
+                ...subscriptionListApiParams,
+                channel_id: '',
+            });
+        } else {
+            setSubscriptionListApiParams({
+                ...subscriptionListApiParams,
+                channel_id: currentChannelId,
+            });
         }
+    };
 
+    // Reset the state when the component is unmounted
+    useEffect(() => {
         return () => {
             handleResetProjectDetails();
         };
     }, []);
 
     useEffect(() => {
-        if (subscriptionData) {
-            if (showAllSubscriptions) {
-                setSubscriptionList(subscriptionData);
-            } else {
-                setSubscriptionList(getCurrentChannelSubscriptions(subscriptionData, currentChannelId));
-            }
-        }
-    }, [subscriptionData, showAllSubscriptions]);
+        fetchSubscriptionList();
+    }, [subscriptionListApiParams]);
 
-    // Update subscription list on switching channels
     useEffect(() => {
-        if (subscriptionData) {
-            setShowAllSubscriptions(false);
-            setSubscriptionList(getCurrentChannelSubscriptions(subscriptionData, currentChannelId));
+        if (subscriptionListApiParams.channel_id !== currentChannelId) {
+            setSubscriptionListApiParams({
+                ...subscriptionListApiParams,
+                channel_id: currentChannelId,
+            });
         }
     }, [currentChannelId]);
```

Comment:
Now we can modify this like below - 
```
useEffect(() => {
 setShowAllSuggestions(false);
}, [currentChannelId]);
```

Reply:
this led to calling API to fetch the subscription list twice, but that is handled

Reply:
I think we could've handled it by removing this `useEffect` and changing the `showAllSubscriptions` state inside `useMemo` (I'm guessing that it could work like below).

```
 const subscriptionListApiParams: FetchSubscriptionList = useMemo<FetchSubscriptionList>(() => {
        if (currentChannelId !== subscriptionListApiParams.channel_id) {
            setShowAllSubscriptions(false);
        }
        return {
            project: projectName,
            channel_id: showAllSubscription && currentChannelId === subscriptionListApiParams.channel_id ? '' : currentChannelId,
        };
    }, [projectName, currentChannelId, showAllSubscription]);
```

Reply:
this will not work because here
we are accessing `subscriptionListApiParams.channel_id` before initialization

also, if we do handle this, another case would be that when the toggle is on and we try to turn it off then `subscriptionListApiParams` will be modified twice once for `setShowAllSubscriptions(false)` and other for `channel_id: showAllSubscription && currentChannelId === subscriptionListApiParams.channel_id ? '' : currentChannelId` which will again call API twice